### ISSUE TITLE
Create new ResourceStat model/table for Analytics

### ISSUE THREAD

**Issue by mcritchlow** [samvera/hyrax#2735]
### Descriptive summary
Table structure:
```
ActiveRecord::Schema.define(version: 20_171_026_154_256) do
  create_table 'resource_stats', id: :serial, force: :cascade do |t|
    t.datetime 'date'
    t.integer 'downloads'
    t.integer 'pageviews'
    t.integer 'unique_visitors'
    t.integer 'returning_visitors'
    t.string 'resource_id'
    t.string 'resource_type'
    t.datetime 'created_at', null: false
    t.datetime 'updated_at', null: false
    t.integer 'user_id'
  end
  # Site-wide stats: 
  add_index :resource_stats, :date
  # Work/File pageviews by user X on date Y
  add_index :resource_stats, [ :user_id, :date, :resource_id, :resource_type ] 
end
```

To do:

- [ ] Create DB migration
- [ ] Define proposal for indexes, ensure we're not missing any important combinations
- [ ] Create initial `ResourceStat` model
- [ ] Add initial test coverage

### Rationale

This will be a small PR to make sure we're all on the same page. Following the merge of this PR, I'll work on #2736 

### Related work

#2736

**Comment by mcritchlow** [comment]
I've got a first pass at the model with a very basic test asserting that we always have a `date` ready to go. Once our branch is rebased against master I'll rebase my branch and put up a PR. Starting to think about the scope/query methods we'll need.

**Comment by mcritchlow** [comment]
Fixed in #2750
